gammapy/spectrum/energy.py

```diff
@@ -0,0 +1,204 @@
+
+    Parameters
+    ----------
+
+    energy : `~numpy.ndarray`
+        Energy
+    unit : `~astropy.units.UnitBase`, str
+        The unit of the values specified for the energy.  This may be any
+        string that `~astropy.units.Unit` understands, but it is better to
+        give an actual unit object.
+
+    """
+
+    def __init__(self, energy, unit=None):
+
+        if isinstance(energy, Quantity):
+            energy = energy.value
+            unit = energy.unit
+
+        if isinstance(unit, u.UnitBase):
+            pass
+        elif isinstance(unit, basestring):
+            unit = u.Unit(unit)
+        elif unit is None:
+            raise UnitsError("No unit was specified in Energy initializer")
+
+        self._unit = unit
+        self._value = energy
+
+    @staticmethod
+    def equal_log_spacing(emin, emax, nbins):
+        """Create Energy with equal log-spacing.
+
+        The unit will be taken from emax
+
+        Parameters
+        ----------
+        emin : `~astropy.units.Quantity`
+            Lowest energy bin
+        emax : `~astropy.units.Quantity`
+            Highest energy bin
+        bins : int
+            Number of bins
+
+        Returns
+        -------
+        Energy
+        """
+
+        if not isinstance(emin, Quantity) or not isinstance(emax, Quantity):
+            raise ValueError("Energies must be Quantities")
+
+        x_min, x_max = np.log10([emin.value, emax.value])
+        energy = np.logspace(x_min, x_max, nbins)
+        energy = Quantity(energy, emax.unit)
+
+        return Energy(energy)
+
+    @staticmethod
+    def from_fits(hdulist):
+        """Read ENERGIES fits extension.
+
+        Parameters
+        ---------
+        hdu_list : `~astropy.io.fits.HDUList` 
+            HDU list with ``ENERGIES`` extensions.
+
+        Returns
+        ------
+        Energy
+        """
+
+        energy = Quantity(hdulist['ENERGIES'].data['Energy'], 'MeV')
+        return Energy(energy)
+
+    def to_fits(self, **kwargs):
+        """Write ENERGIES fits extension
+
+        Returns
+        -------
+        hdu_list : `~astropy.io.fits.BinTableHDU`
+            ENERGIES fits extension
+
+
+        """
+
+        col1 = fits.Column(name='Energy', format='D', array=self.value)
+        cols = fits.ColDefs([col1])
+        return fits.BinTableHDU.from_columns(cols)
+
+    def info(self):
+        # is this really necessary? what's the benefit towards just typing the
+        # name of the instance?
+        pass
+
+class EnergyBounds(Quantity):
+
+    """Energy bin edges
+
+    Stored as "EBOUNDS" FITS table extensions.
+
+    Parameters
+    ----------
+
+    energy : `~numpy.ndarray`
+        Energy
+    unit : `~astropy.units.UnitBase`, str
+        The unit of the values specified for the energy.  This may be any
+        string that `~astropy.units.Unit` understands, but it is better to
+        give an actual unit object.
+
+    """
+
+    def __init__(self, energy, unit=None):
+
```

Comment:
Your link is to version 0.2.1 of the Astropy docs, where quantity didn't exist.

These days, Angle does sub-class quantity:
http://astropy.readthedocs.org/en/latest/api/astropy.coordinates.Angle.html

If you click on "source" you'll see that it doesn't define an `__init__`, but a complex, 30 line `__new__`.

Here's some high-level docs how to sub-class quantity:
http://astropy.readthedocs.org/en/latest/units/quantity.html#subclassing-quantity

I've never sub-classed quantity and don't have time to learn and play with this now.
I don't know if your current constructor works OK, but given how Angle does it, my guess is that it doesn't work properly for all cases (i.e. initialised via arrays, other quantities, lists, scalars, ...).

So the two options for now are:
1. learn more about quantity and subclassing it and add quite a few tests to show that it works. you'll learn a lot about Python, but it will take time.
2. punt on the complexity of sub-classing quantity, and instead use composition, i.e. store the energy array as a `.data` or `.quantity` data member.

I'm sorry that I don't have time to think about this more and look into this myself, until Wednesday next week you're on your own (or ask Axel for advice).